### ISSUE TITLE
fullscreen doesn't cover entire screen, panel-bar remains visible in linux cinnamon / xfce

### ISSUE THREAD

**Issue by danieljpetersen** [ggez/ggez#606]
This is on version 0.5.0-rc.1 while running the [graphics_settings](https://github.com/ggez/ggez/blob/master/examples/graphics_settings.rs) example.  See screenshot

![Screenshot from 2019-03-16 16-57-20](https://user-images.githubusercontent.com/5576311/54481647-d2356480-480d-11e9-8440-c61bec41e363.png)

That demo starts in windowed mode.  I tried setting it to fullscreen at the start and the same result happens.  

I also tested it on Mate, where it did indeed cover the entire screen.  Though there was a separate issue there where my entire computer would freeze after toggling from fullscreen back to windowed.

**Comment by icefoxen** [comment]
Gotta say, that's pretty fascinating.  Thank you for the bug report.  What OS, GPU and graphics drivers are you on?  Can't reproduce the Mate freeze on my machine (Debian, NVidia GTX 660 with NVidia drivers), I'll try Cinnamon and XFCE.

**Comment by danieljpetersen** [comment]
OS:  Linux Mint 19.1.  
GPU:  Geforce 780 TI
Drivers:  415.27

**Comment by Doimhneacht** [comment]
Hi. I'm experiencing the same problem.

ggez version: 0.5.1
OS: Arch Linux
KDE Plasma Version: 5.16.4
KDE Frameworks Version: 5.60.0
Qt Version: 5.13.0
GPU: GeForce 940MX
NVidia Driver Version: 430.40

**Comment by icefoxen** [comment]
Yeah, I just tried in Cinnamon and the panel doesn't get covered there either.  So, I think we can safely consider this confirmed.  It DOES appear to work properly in true fullscreen mode instead of windowed fullscreen.  This was using both winit 0.19.1 and 0.19.2.  Need to see if it reproduces with just winit examples.

**Comment by icefoxen** [comment]
Okay, `winit` 0.20 fixes this bug.  `winit` 0.19 doesn't actually support windowed fullscreen as its own thing, you just make a big window without decorations.  Apparently this isn't always enough.  Not sure if there's a better way to coax `winit` 0.19 into doing it.

**Comment by Athorus** [comment]
Same issue on Windows 10.

**Comment by icefoxen** [comment]
Reproduced on Windows 10, more or less.  Sometimes you have to switch to windowed fullscreen and back a few times to get it to show up.

winit 0.19.4 released on October 16, 2019, and I've never seen this bug before on Windows, so I'm suspicious it's a new feature of winit.

**Comment by Athorus** [comment]
With the following code, fullscreen works on Windows 10.

```rust
let window = graphics::window(ctx);
let monitor = window.get_current_monitor();
window.set_fullscreen(Some(monitor));
```

PS: winit = "0.19.5" in my cargo.toml

**Comment by Vixeliz** [comment]
Can this be closed now since 0.20 fixed this and has long since been used in ggez?

**Comment by danieljpetersen** [comment]
> Can this be closed now since 0.20 fixed this and has long since been used in ggez?

I originally opened this issue. Reading this prompted me to re-test and it's still not 100% fixed (tested against master).

Running the graphics_settings example
- Starts out windowed
- Press F to toggle fullscreen. The window does not cover the menubar
- Press F to toggle back to windowed
- Press F to go back to fullscreen. It now covers the menubar
- Press F again to toggle back to windowed -- it remains fullscreen

**Comment by Vixeliz** [comment]
Kk thanks for your comment and testing :)
